sway-lib-std/src/crypto/ed25519.sw

```diff
@@ -0,0 +1,216 @@
+library;
+
+use ::alloc::alloc_bytes;
+use ::b512::B512;
+use ::bytes::Bytes;
+use ::convert::{From, TryFrom};
+use ::crypto::{message::Message, public_key::PublicKey, signature_error::SignatureError};
+use ::hash::*;
+use ::result::Result::{self, *};
+use ::option::Option::{self, *};
+
+/// An ed25519 signature.
+pub struct Ed25519 {
+    /// The underlying raw `[u8; 64]` data of the signature.
+    bits: [u8; 64],
+}
+
+impl Ed25519 {
```

Comment:
I notice that `new`, `bits` and `verify` are implemented for all 3 signature types, and `recover`, `address`, `evm_address`, `verify_address` and `verify_evm_address` are implemented for both secp256k1 and secp256r1.

Would there be value in defining trait with those methods, which the signature types can then implement?

Reply:
These are not possible with the Ed25519 cryptography. Ed25519 only allows for verification as it currently stands https://docs.fuel.network/docs/specs/fuel-vm/instruction-set/#ed19-eddsa-curve25519-verification